server/store/datastore/task_test.go

```diff
@@ -39,19 +38,10 @@ func TestTaskList(t *testing.T) {
 		t.Error(err)
 		return
 	}
-	if got, want := len(list), 1; got != want {
-		t.Errorf("Want %d task, got %d", want, got)
-		return
-	}
-	if got, want := list[0].ID, "some_random_id"; got != want {
-		t.Errorf("Want task id %s, got %s", want, got)
-	}
-	if got, want := list[0].Data, "foo"; string(got) != want {
-		t.Errorf("Want task data %s, got %s", want, string(got))
-	}
-	if got, want := list[0].DepStatus, map[string]string{"test": "dep"}; !reflect.DeepEqual(got, want) {
-		t.Errorf("Want task data %s, got %s", want, got)
-	}
+	assert.Len(t, list, 1, "Expected one task in list")
+	assert.EqualValues(t, "some_random_id", list[0].ID)
+	assert.EqualValues(t, "foo", list[0].Data)
```

Comment:
```suggestion
	assert.Equal(t, "some_random_id", list[0].ID)
	assert.Equal(t, "foo", list[0].Data)
```

Reply:
As the `Data` attribute is type byte array the `assert.Equal` will not work in that case. See the unit test output:
```
=== RUN   TestTaskList
    task_test.go:43: 
        	Error Trace:	/Users/lukasz/projects/woodpecker/server/store/datastore/task_test.go:43
        	Error:      	Not equal: 
        	            	expected: string("foo")
        	            	actual  : []uint8([]byte{0x66, 0x6f, 0x6f})
        	Test:       	TestTaskList
--- FAIL: TestTaskList (0.00s)
```
So, I just changed the assert for `ID` which will work with `Equal`.

Reply:
jap, the string casting will work as well